summit/roiGeneration.m

```diff
@@ -1,340 +1,398 @@
+        
+        % store all of the coordinates of the array bounds to the struct
+        struct.array_bounds = array_bounds;
+        
+        
+        % Construct a questdlg to ask the user if they are happy with their
+         % well selection
+        choice = questdlg('Are you happy with your well selections?', ...
+        'Well selections for array boundaries', ...
+        'Yes','No','Yes');
+        
+        % Handle response
+        switch choice
+            
+            case 'Yes';
+                disp([choice 'Great, let''s keep going then!'])
+            
+            case 'No';
+                disp([choice 'That''s okay, try again!'])
+                test = 1;
+        end
+     
+    % check whether the user selected array boundaries are correct    
+    if (x_rightwell<x_leftwell || y_bottomwell<y_topwell)        
+        test = 1;
+        
+        title('Oh no! We detected you selected the wells in the wrong order. Please try again. Press any key to continue')
+        pause()
+    else
+        test = 0;
+    end
+  end
+    transform=1;
+end
+
+
+% Determine number of wells per row
+wells_per_row = round((array_bounds(2,1) - array_bounds(1,1)) / horzspacing)+1;
+struct.wells_per_row = wells_per_row;
+
+% Determine number of rows
+rows = round((array_bounds(4,2)-array_bounds(3,2)) / vertspacing)+1;
+
+struct.rows = rows;
+
+% Determine total number of wells
+total_wells = wells_per_row * rows;
+
+
+% for loop to fill in the 3D matrix with ROIs from the image (proceeds row by row of the microwell array from left to right)
+% pre-allocate 3D matrix with zeros
+mat = zeros(vertspacing, horzspacing, total_wells);
+
+for i = 1:rows
+    for j = 1:wells_per_row
+        
+        % determine z-coordinate for the current ROI
+        z = (wells_per_row) * (i-1)+j;
+        
+        % set row start and end boundaries 
+        row_start = (round(array_bounds(1,1)) - horzspacing/2) + ((j-1)*horzspacing);
+        row_end = row_start + horzspacing;
+        
+        % set column start and end boundaries
+        col_start = (round(array_bounds(3,2)) + ((i-1)*vertspacing));
+        col_end = col_start + vertspacing;
+        
+        %generate lines that span the x and y coordinates of all the ROIs
+        %to overlay over image to show the ROIs
+        x = row_start:1:(row_end - 1);
+        y = repmat(col_start, 1, length(x));
+        y2 = col_start:1:(col_end);
+        x2 = repmat((row_end), 1, length(y2));
+        
+        % fill the matrix with the image pixels within the current ROI
+        % boundaries
+        mat(: ,: ,z) = b(col_start:(col_end - 1), row_start:(row_end - 1));
+        
+        % plot the ROI grid overlay on the image
+        plot(x', y', 'Color', 'r', 'LineStyle','-');
+        plot(x', y', 'Color', 'r', 'LineStyle',':');
+        plot(x2', y2', 'Color', 'r', 'LineStyle','-');
+        plot(x2', y2', 'Color', 'r', 'LineStyle',':');
```

Comment:
When I resolved the merge conflict, I kept these lines from the roiGenerationEdit branch